playground/2D_explicit.py

```diff
@@ -0,0 +1,118 @@
+# We will use the 5-point Laplacian aprroximation stencil explicit method 
+# below to simulate the 2D Heat Equation
+import numpy as np
+import matplotlib.pyplot as plt
+
+from matplotlib.animation import FuncAnimation
+
+# Global Constants
+xLength = 50
+yLength = 50
+maxTime = 1
+diffusivityConstant = 1
+numPointsSpace = 200
+numPointsTime = 60
+
+# Length Vector plotted on x-axis and y-axis
+xDomain = np.linspace(0, xLength, numPointsSpace)
+yDomain = np.linspace(0, yLength, numPointsSpace)
+# Time Vector plotted on z-axis
+timeDomain = np.linspace(0, maxTime, numPointsTime)
+
+# Step-sizes- dt, dx, dy
+timeStepSize = timeDomain[1] - timeDomain[0]
+spaceStepSize = xDomain[1] - xDomain[0]
+
+# Stability condition: dt <= dx^2/(2*k)
+err_dt= np.abs((spaceStepSize**2)/(2*diffusivityConstant))
+print(err_dt)
+assert(timeStepSize <= err_dt)
+
+# Calculate lambda constant for x and y (they're the same if dx = dy)
+lambdaConstant = (diffusivityConstant * timeStepSize) / (spaceStepSize**2)
+# Stability condition: kdt/dx^2 <= 0.5
+assert(lambdaConstant <= 0.5) 
+print(lambdaConstant)
+
+# lambda functions for initial condition and boundary condtions
+# 2 sin x sin 2y + 3 sin 4x sin 5y + t
+u0 = lambda x, y: 2 * np.sin(x) * np.sin(2*y) + 3 * np.sin(4*x) * np.sin(5*y)
+left = lambda t, y: 2 * np.sin(0) * np.sin(2*y) + 3 * np.sin(4*0) * np.sin(5*y) + t
+right = lambda t, y: 2 * np.sin(xLength) * np.sin(2*y) + 3 * np.sin(4*xLength) * np.sin(5*y) + t
+bottom = lambda t, x: 2 * np.sin(x) * np.sin(2*0) + 3 * np.sin(4*x) * np.sin(5*0) + t 
+top = lambda t, x: 2 * np.sin(x) * np.sin(2*yLength) + 3 * np.sin(4*x) * np.sin(5*yLength) + t
+
+
+# error assertion for intial & boundary conditions
+err1_u0 = np.abs(u0(0, 0) - left(0, 0))
+assert(err1_u0 < 1e-12)
+err2_u0 = np.abs(u0(0, yLength) - right(0, 0))
+assert(err2_u0 < 1e-12)
+err3_u0 = np.abs(u0(0, 0) - bottom(0, 0))
+assert(err3_u0 < 1e-12)
+err4_u0 = np.abs(u0(0, xLength) - top(0, 0))
+assert(err4_u0 < 1e-12)
+
+def initMatrix():
```

Comment:
that's OK like this for now, but eventually all function will need to have documentation